forc-pkg/src/pkg.rs

```diff
@@ -1957,6 +1962,56 @@ pub fn compile(
         warnings,
         metrics,
     };
+
+    #[cfg(feature = "profile")]
+    {
```

Comment:
Can we move the code between the { } braces into a function and then call it here?